### DISCUSSION TITLE
Please tell me where is the error in the code

### DISCUSSION THREAD

**Discussion by boubekeur27** [yalmip/YALMIP#1344] [General discussion]
A = [-2 1; 1 -1];
E = [1 8; 1 8];
B= [1 ; -1];
%alpha = 1.2;
n=size(A,1);
n=size(E,1);
[n1,m]=size(B);
X= sdpvar(n, n);
Y= sdpvar(n1, n);
LMI = [2*sin(3/4*pi)*A'*(X*E+B*Y'), zeros(n,n);
    zeros(n,n) , 2*sin(3/4*pi)*A'*(X*E+B*Y')];

const = [LMI<0, X>0,Y>0];
solvesdp(const);

X = double(X);
Y = double(Y);

**Comment by johanlofberg** [comment]
I already told you that it is not an LMI as you don't have a symmetric object. Lyapunov/stability theory is always in the form A'*X+X*A etc which your's clearly fails to satisfy. Hence read the theory again.

Second, I gave you links to read, and you have not read them as there is no use of a function called solvesdp there (as there is no such function officially at all in YALMIP, and there is no official function called double either) and it is also explained that strict inequalities are not supported

**Reply by boubekeur27** [reply]
Thanks prof,
In fact, I want to solve this inequality in the following theorem
[LMI2.pdf](https://github.com/yalmip/YALMIP/files/13253138/LMI2.pdf)

**Reply by johanlofberg** [reply]
yes, and you are not applying the sym operator which clearly is used there, i.e. sym(z) means z+z'

**Comment by boubekeur27** [comment]
Thanks prof,
In fact, I want to solve this inequality in the following theorem

*Theorem :*
The system is admissible if and only if there exist matrices X>0 and Y
satisfying
[image: image.png]
[image: image.png]
and let :
[image: image.png]



Le ven. 3 nov. 2023 à 17:39, Johan Löfberg ***@***.***> a
écrit :

**Comment by boubekeur27** [comment]
Ok, thanks.
Can you help me with that LMI ?

Le ven. 3 nov. 2023 à 19:40, Johan Löfberg ***@***.***> a
écrit :

**Reply by johanlofberg** [reply]
You already coded it, except you have to apply the symmetry operator, use non-strict inequality (and possible then add something which avoids trivial 0 solution), and the correct function for solving it.

**Comment by boubekeur27** [comment]
Ok, thanks for the clarification.

Le sam. 4 nov. 2023 à 12:38, Johan Löfberg ***@***.***> a
écrit :
